ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -1007,7 +1007,10 @@ void checkAndMaybeComplete() {
       if (progressTracker.isDone()) {
         if (progressTracker.hasSucceeded() && !retainChunkExceptionOnSuccess) {
           chunkException = null;
-        } else if (chunkOperationTracker.hasFailedOnNotFound()) {
+        } else if (chunkOperationTracker.maybeFailedDueToOfflineReplicas()) {
```

Comment:
1/ How did you stumble upon this ?
2/ What do we return today for this case - 404 ?
3/ Are we returning 503 if _all_ replicas are offline including the ones in remote DC ?

Reply:
1. It happened in Samza GCN. Will ping you the ticket.
2. Currently we returned 404.
3. We plan to return 503 in different cases. In the PR, we added one case, if we only have 0 or  "eligible replicas in the originating DC", we will return 503.